src/pygambit/game.pxi

```diff
@@ -1122,14 +1122,14 @@ class Game:
 
         resolved_nodes = (
                        [self._resolve_node(n, "append_move") for n in nodes]
-                       if hasattr(nodes, "__iter__")
+                       if (hasattr(nodes, "__iter__") and not isinstance(nodes, str))
                        else [self._resolve_node(nodes, "append_move")]
         )
 
         if any(len(rn.children) > 0 for rn in resolved_nodes):
             raise UndefinedOperationError("append_move(): `node` must be a terminal node")
 
-        if hasattr(nodes, "__iter__") and (len(nodes) != len(set(nodes))):
+        if hasattr(nodes, "__iter__") and any(nodes.count(n) > 1 for n in nodes):
```

Comment:
This misses the point of the previous error cases.

The error here is that `resolved_node` should be used, not `nodes`.